projects/step-frontend/src/lib/modules/grid/components/token-list/token-list.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnDestroy } from '@angular/core';
+import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
+import { AJS_MODULE, Mutable, TableLocalDataSource, GridService, TokenWrapper } from '@exense/step-core';
+import { Observable, BehaviorSubject, switchMap, shareReplay, tap } from 'rxjs';
+import { FlatObjectStringFormatPipe } from '../../services/flat-object-format.pipe';
+import { TokenTypeComponent } from '../token-type/token-type.component';
+
+type InProgress = Mutable<Pick<TokenListComponent, 'inProgress'>>;
+
+@Component({
+  selector: 'step-token-list',
+  templateUrl: './token-list.component.html',
+  styleUrls: ['./token-list.component.scss'],
+})
+export class TokenListComponent implements OnDestroy {
+  readonly inProgress: boolean = false;
+
+  private tokenRequestSubject$ = new BehaviorSubject<any>({});
+  readonly tokenRequest$: Observable<TokenWrapper[]> = this.tokenRequestSubject$.pipe(
+    tap((_) => ((this as InProgress).inProgress = true)),
+    switchMap((_) => this._gridService.getTokenAssociations()),
+    tap((_) => ((this as InProgress).inProgress = false)),
+    shareReplay(1)
+  );
+
+  readonly searchableToken$ = new TableLocalDataSource(this.tokenRequest$, {
+    searchPredicates: {
+      id: (element, searchValue) => element.token!.id!.toLowerCase().includes(searchValue.toLowerCase()),
+      type: (element, searchValue) =>
+        TokenTypeComponent.TYPE_LABEL_TRANSLATIONS[element.token!.attributes!['$agenttype']]!.toLowerCase().includes(
+          searchValue.toLowerCase()
+        ),
+      agent: (element, searchValue) => element.agent!.agentUrl!.toLowerCase().includes(searchValue.toLowerCase()),
+      attributes: (element, searchValue) =>
+        FlatObjectStringFormatPipe.format(element.token!.attributes!).includes(searchValue.toLowerCase()),
+      state: (element, searchValue) => searchValue.toUpperCase().includes(element.state!.toUpperCase()),
+      // prettier-ignore
+      //@ts-ignore
```

Comment:
Why do we use so many ts-ignore in this file? I looked at the errors and it seems we could just add the missing properties to the models or am I missing something?

Reply:
introduced a new type that has all the fields that are returned by the API, so got rid of ts-ignore :+1: